### PR TITLE
List unsubmitted job listings in job dashboard and allow completion

### LINKED ISSUE #1644: Allow logged in users to complete and publish their drafts
Replaces #1634
Replaces #964

Allow logged in users to continue draft job listings from the `[job_dashboard]` shortcode. This should allow them to pick up an unsubmitted job listing without having to make the cookies more than a session cookie.

### PR DESCRIPTION
**Pull request by jom**
Fixes #1644

#### Changes proposed in this Pull Request:
I thought this was going to be more difficult. Maybe I'm overlooking something, but this seems to work for me. Have just done some testing but will do more later.

* For `draft` (started in WP Admin) and `preview` (started in frontend) job listings, list them in the `[job_dashboard]` and allow users to continue their submission.
* This does not allow for continuation of listings stuck in `pending_payment` (WC Paid Listings). We'll need to build off this PR in that plugin.
* It shows the post status of `preview` as `Preview`, but that might be confusing. We could map the human readable label for `preview` to `draft`'s label in `get_the_job_status()`. 
* Folks who want to disable this can filter `job_manager_get_dashboard_jobs_args` and remove the post statuses. 

#### Testing instructions:

* Using both WP admin and then again through `[submit_job_listing]` form, start some listings. In WP admin, just save draft. In frontend, stop at Preview stage.
* Visit `[job_dashboard]` page. 
* Click `Continue Submission` under either listing. 
* Should allow you to move to Preview stage and post (assuming no WCPL or similar plugin is active).

### CODE REVIEW

**Comment by vetpetjobs** [comment]
Suggest test with complete log out and log back in - this logout issue was part of the original support request that triggered this item

**Comment by donnapep** [comment]
For a job added via WP admin, when I continue the submission and submit, I'm taken to an empty _Post a Job_ screen with no confirmation message, and the job listing remains in Draft status.

Also, if I had a previous job created in WP Admin that's in Draft status, and then I switch to this branch, that job does not have the "Continue Submission" link. That may be expected behaviour though.

Front-end jobs seem to work. After submitting, I'm taken to the _Post a Job_ screen, but I see the message "Job submitted successfully. Your listing will be visible once approved." The status is changed to "Pending approval".

**Comment by jom** [comment]
> Also, if I had a previous job created in WP Admin that's in Draft status, and then I switch to this branch, that job does not have the "Continue Submission" link. That may be expected behaviour though.

I wasn't able to replicate this issue. I created a draft on `master`, switched to this branch, and the **Continue Submission** link appeared under job dashboard.

> For a job added via WP admin, when I continue the submission and submit, I'm taken to an empty Post a Job screen with no confirmation message, and the job listing remains in Draft status.

I was able to replicate this issue. Fixed in cf0c5a3ce666f70f6a1f305ba1f5463af718b764

**Review by donnapep — APPROVED** [review]

**Pull request merged**
